### PR TITLE
Add health endpoint to backend websocket API

### PR DESCRIPTION
**Pull request by amdprophet**
## What is this change?

Adds an unauthenticated `/health` endpoint to the backend websocket API.

## Why is this change necessary?

See #2773 

## Does your change need a Changelog entry?

Yes.

## Do you need clarification on anything?

No.

## Were there any complications while making this change?

I initially misunderstood how the actions controller for health worked.

## Have you reviewed and updated the documentation for this change? Is new documentation required?

Yes, new documentation is required.

## How did you verify this change?

I wrote a test to ensure that the `/health` endpoint on the agentd websocket port returns a 200 status.

## Is this change a patch?

No.

### CODE REVIEW

File: backend/agentd/agentd.go [hunk 1]
```diff
@@ -141,8 +149,15 @@ func New(c Config, opts ...Option) (*Agentd, error) {
...
 	router := mux.NewRouter()
-	router.HandleFunc("/", a.webSocketHandler)
-	router.Use(authenticate, authorize, entityLimit, agentLimit)
+
+	healthRouter := routers.NewHealthRouter(
+		actions.NewHealthController(a.store, a.client.Cluster, a.etcdClientTLSConfig),
+	)
+	healthRouter.Mount(router)
+
+	route := router.NewRoute().Subrouter()
```
Comment (ccressent) [inline backend/agentd/agentd.go:158]: Not sure I understand this line.

**Review by ccressent — APPROVED** [review]
I did not test this PR but it seems reasonable to me. I have a few questions though:

- #2773 mentions also UI and "backend" ; that `/health` only covers agentd's websocket though?
- Don't we already have existing health endpoints? Can we not instead relax them a bit so an unauthenticated request can still get barebone information (like just a 200 OK)?

Those are just general interest questions I unfortunately do not have the time to go answer myself right now.

File: backend/agentd/agentd.go (hunk 1 above)
Comment (amdprophet) [reply backend/agentd/agentd.go:158]: This creates a new subrouter that apply the authentication & authorization middleware to.

**Comment by amdprophet** [comment]
> * #2773 mentions also UI and "backend" ; that `/health` only covers agentd's websocket though?
I think UI and backend (API) are already covered.

> * Don't we already have existing health endpoints? Can we not instead relax them a bit so an unauthenticated request can still get barebone information (like just a 200 OK)?
User(s) requested a way to verify if the agent transport (websocket) is running. As per @portertech's comment (https://github.com/sensu/sensu-go/issues/2773#issuecomment-696269962) I reused the code for the code that the API's `/health` route is using.

**Pull request merged**